docs: update external docs with a better structure (#14718)
rook/rook: Documentation/CRDs/Cluster/external-cluster/upgrade-external.md

> # Upgrade external cluster

Reviewer: ```suggestion
# External Cluster Upgrades
```

> 

Reviewer: Could you provide an overview about the upgrades? It will help to understand what type of upgrade is being referred to, instead of only directly diving into details about caps.

Reviewer: Still planning on this or is it somewhere else? I was thinking it would be before this heading

    ```console
    python3 create-external-cluster-resources.py --upgrade
    ```

2. If the consumer cluster has restricted caps

    Restricted users created using `--restricted-auth-permission` flag need to pass mandatory flags: '`--rbd-data-pool-name`(if it is a rbd user), `--k8s-cluster-name` and `--run-as-user`' flags while upgrading, in case of cephfs users if you have passed `--cephfs-filesystem-name` flag while creating CSI users then while upgrading it will be mandatory too. In this example the user would be `client.csi-rbd-node-rookstorage-replicapool` (following the pattern `csi-user-clusterName-poolName`)

    ```console
    python3 create-external-cluster-resources.py --upgrade --rbd-data-pool-name replicapool --k8s-cluster-name rookstorage --run-as-user client.csi-rbd-node-rookstorage-replicapool
    ```

    !!! note
        1) An existing non-restricted user cannot be converted to a restricted user by upgrading.
        2) The upgrade flag should only be used to append new permissions to users. It shouldn't be used for changing a CSI user already applied permissions. For example, be careful not to change pools(s) that a user has access to.

> # Upgrade cluster to utilize new feature

Reviewer: ```suggestion
## Upgrade cluster to utilize a new feature
```

Some Rook upgrades may require re-running the import steps, or may introduce new external cluster features that can be most easily enabled by re-running the import steps.

To re-run the import steps with new options, the python script should be re-run using the same configuration options that were used for past invocations, plus the configurations that are being added or modified.

Starting with Rook v1.15, the script stores the configuration in the external-cluster-user-command configmap for easy future reference.

* arg: Exact arguments that were used for for processing the script. Argument that are decided using the Priority: command-line-args > config.ini file values > default values.

> ## Example `external-cluster-user-command` ConfigMap:

Reviewer: ```suggestion
### Example `external-cluster-user-command` ConfigMap:
```

# External Cluster Upgrades

> When upgrading an external cluster, Ceph and Rook versions will be updated independently. During the Rook update, we also need to update the external provider cluster connection with any settings and permissions for new features.

Reviewer: nit: remove usage of "we"
```suggestion
When upgrading an external cluster, Ceph and Rook versions will be updated independently. During the Rook update, the external provider cluster connection also needs to be updated with any settings and permissions for new features.
```

# External Cluster Upgrades

> Upgrading external cluster is little tricky, We need to upgrade the Ceph cluster and Rook cluster seperalty with their new versions, Plus upgrading the versions we also need to update the external provider cluster connection to consumer Rook cluster.

Reviewer: How about this?
```suggestion
When upgrading an external cluster, Ceph and Rook versions will be updated independently. During the Rook update, we also need to update the external provider cluster connection with any settings and permissions for new features.
```

> To upgrade the provider ceph cluster to use the latest ceph user permissions, which will be required for the rook reconciliation, based on the new implementation added to the Rook.

Reviewer: Does my suggestion above also cover this concept of new permissions and properties? Then no need for this paragraph.
```suggestion
```

> ## Upgrade the cluster to consume latest ceph user caps(mandatory)

Reviewer: For proper rendering, we usually need a blank link before and after titles
```suggestion

## Upgrade the cluster to consume latest ceph user caps (mandatory)
```
